### DISCUSSION TITLE
Integrate lit with jest

### DISCUSSION THREAD

**Discussion by mohinderps** [lit/lit#4450] [Q&A]
We have built our web component library with lit. We used boilerplate provided by Vite which uses Vitest to test the components. This is working fine.

When we use the library in our vue application, we are able to import the component and use it. But when we run the test cases, written using jest, the test fails because the component does not get rendered.

For example for a component gc-button, the underlying DOM is not rendered.

**Comment by augustjk** [comment]
It's hard to say without a reproduction on why your test setup isn't working. The difficulty with Jest is that it's running in Node and you'd require some kind of DOM emulation to get custom element rendering to work, usually jsdom or happydom. I think those do offer some level of custom element support but there might be some configuration required. There are other variables like how your source code is processed before being provided to the test.
